Website/Portals/_default/HotcakesViews/_default/Scripts/Checkout.js

```diff
@@ -866,6 +879,61 @@
         $("table.totaltable").attr("class", "table table-striped table-hover totaltable");
     }
 
+    async function CreatePaymentMethod(clientSecret) {
+        var status = await checkPaymentStatus(clientSecret)
+        if (status === "requires_payment_method") {
+            var cardNumber = $("#cccardnumber").val();
+            var cvc = $("#ccsecuritycode").val();
+            var expMonth = $("#ccexpmonth").val();
+            var expYear = $("#ccexpyear").val();
+            var paymentIntent = $("#PaymentIntentId").val();
+            var pm = "";
+            if (cardNumber && cvc && expMonth && expYear) {
+                var reqUrl = hcc.getServiceUrl("checkout/AttachPaymentMethod");
+                $.post(reqUrl, { "CardNumber": cardNumber, "Cvc": cvc, "ExpMonth": expMonth, "ExpYear": expYear, "PaymentIntentId": paymentIntent }, null, "json")
+                    .done(function (data) {
+                        console.log(data);
+                        stripe
+                            .retrievePaymentIntent(clientSecret)
+                            .then(async function (result) {
+                                if (result.paymentIntent) {
+                                    if (result.paymentIntent.status == "requires_action") {
+                                        var result = await stripe.confirmCardPayment(clientSecret);
+
+                                        if (result.paymentIntent) {
+                                            console.log(result);
+                                            Addresses.saveForm();
+                                        } else {
+                                            Addresses.saveForm();
+                                        }
+                                    } else {
+                                        Addresses.saveForm();
+                                    }
+                                } else {
+                                    Addresses.saveForm();
+                                }
+                            });
+                    })
+                    .fail(function (xhr, status, error) {
+                        console.log("xhr: " + xhr);
```

Comment:
It is a security violation in any security scan to emit full-text error messages to the browser, unless they are emitted only to privileged users.  This is emitted to everyone.  Instead, the error should be trapped, handled, and logged.  

It's fine to emit these messages during development, but now we need to do something else before we push this to the repo.  

So, please ensure that if there's an error, it's logged before the JS receives the error.  Then, use the error state to handle the error in a usable way.